algosdk/abi/tuple_type.py

```diff
@@ -275,13 +266,13 @@ def decode(self, tuple_string):
                             "expected before index should have number of bool mod 8 equal 0"
                         )
                     after = min(7, after)
+                    bit = int.from_bytes(
+                        bytestring[array_index : array_index + 1],
+                        byteorder="big",
+                    )
```

Comment:
nit: actually since you're only extracting 1 byte, `bit = bytestring[array_index]` would be sufficient. `bits` might be a better name though